js/modules/export-data.src.js

```diff
@@ -354,12 +354,26 @@ Highcharts.Chart.prototype.getDataRows = function (multiLevelHeaders) {
             categoryMap: categoryMap,
             dateTimeValueAxisMap: dateTimeValueAxisMap
         };
+    }, 
+    // Create point array depends if xAxis is category
+    // or point.name is defined #13293
+    getPointArray = function (series, xAxis) {
+        if (series.data.filter(function (d) { return d.name; }).length &&
```

Comment:
I don't think that series.data.filter should be as a part of condition. Its unreadable.

I suggest to extract the filter into extra variable i.e. `namedPoints` and then use "namedPoints.length"